src/dd/Simulation.cpp

```diff
@@ -212,34 +246,25 @@ std::map<std::string, std::size_t> sample(const qc::QuantumComputation& qc,
 
 VectorDD simulate(const qc::QuantumComputation& qc, const VectorDD& in,
                   Package& dd) {
-  auto permutation = qc.initialLayout;
-  auto e = in;
+  qc::Permutation permutation = qc.initialLayout;
+  dd::VectorDD out = in;
```

Comment:
Any particular reason for not keeping these as `auto`? (I like the change to refer to this as `out` instead of the generic `e`)

Reply:
Nope. Should have changed that back to `auto`.

This is actually something I wanted to ask you eventually (or your opinion on this topic)!:
The `auto` keyword is used quite extensively in this project. Do you generally prefer `auto` over naming the exact type? In my head the rule of thumb is: use `auto` for lambdas, iterators, and types with long names. However, `qc::Permutation` and `dd::VectorDD` aren't really that many characters. Moreover, when using `auto` any good idea will output the actual type right next to `auto` (so it doesn't really save screen space). Nonetheless, it's very, very, convenient to use `auto`.

Reply:
Just my personal opinion:
The biggest argument is convenience. As you say, IDEs are clever enough by now to figure out the right type and display it appropriately. It saves you from having to think what the actual type should be and what you need to type. That creates a much smoother development experience in my opinion.
I'd generally just avoid `auto` wherever it might add to the clarity of the code to spell out the type.
Or in cases where you actually want a base class pointer but the function returns a pointer from a derived class.
Otherwise I'd almost always go for the auto variant.
The only real downside I can see is when you are not in an IDE and don't have access to all the automatic type hints. Such as when reviewing code on GitHub.

Reply:
okay, thanks!